includes/Core/Key_Metrics/Key_Metrics_Settings.php

```diff
@@ -49,6 +49,27 @@ protected function get_default() {
 		return array();
 	}
 
+	/**
+	 * Merges an array of settings to update. Only existing keys will be updated.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $partial Partial settings array to save.
+	 * @return bool True on success, false on failure.
+	 */
+	public function merge( array $partial ) {
+		$settings = $this->get();
+		$partial  = array_filter(
+			$partial,
+			function ( $value ) {
+				return null !== $value;
+			}
+		);
+		$updated  = array_intersect_key( $partial, $settings );
```

Comment:
@eugene-manuilov thanks for updating this method, however, I notice when testing the endpoint (and as the JSDoc does mention) that only existing keys will be updated, so when trying to POST when the settings have the default value of an empty array, the update is ignored.

My understanding is that the settings will only be updated in response to user input, which means we need to be able to merge them from the default value. With that in mind, I think we need a slight variation on this implementation. Either by specifying the list of allowed keys here instead of using `$settings` to determine the allowed keys, or potentially by updating `get_sanitize_callback` to drop unknown fields.

This also implies there should be an additional test case to test merging when the settings are not yet present.